lib/array_equals.rb

```diff
@@ -1,5 +1,34 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
-end
+  # checks edgecases first, return values will stop method due to short-circuit 
+  
+  if array1 == nil && array2 == nil
+    return true
+  end
+  
+  if array1 == [] && array2 == []
+    return true
+  end
+  
+  if array1 == nil && array2 != nil
```

Comment:
You could just say 

```ruby
if array1.nil?  || array2.nil?
```

Since you know by this point that they're not **both** nil.